### ISSUE TITLE
Training configurations with ResNet18

### ISSUE THREAD

**Issue by dipamgoswami** [GwenLegate/GuidingLastLayerFLPretrain#2]
Hi @GwenLegate,
Can you share the training configurations for using the ResNet18 model for all the datasets?

Thanks!

**Comment by GwenLegate** [comment]
Sorry, I just noticed this issue now! I'll add my response here in case anyone else is wondering the same thing

hyperparameters for resnet18:

- In all cases we use 100 clients with data evenly distributed between them (500 samples each) using a dirichlet distribution parameterized by alpha=0.1.
- data at each client is further partitioned into 90% training and 10% validation
- We use SGD with weight decay 1e-4 for the optimizer
- Since the models are pretrained using batch norm, we use it too even though some studies have indicated batch norm can negatively affect convergence in federated learning
- we do one client epoch and a batch size of 64 in both cases
- we use full client participation for flowers and 30% participation for cifar10

for flowers:

- FedNCM basically has no hyperparameters
- FedNCM+FT has lr=0.0005 and 500 rounds of training
- just FT with a pretrained model uses lr=0.007 and 1500 rounds
- just LP with a pretrained model uses lr=0.003 and 1000 rounds

with cifar:

- FedNCM basically has no hyperparameter
- FedNCM+FT has lr=0.001 and 500 rounds of training
- just FT with a pretrained model uses lr=0.003 and 500 rounds
- just LP with a pretrained model uses lr=0.03 and 500 rounds

If you're using our code you should be able to set these hyperparameters using config.py (the FLSim way) or I coded a command line option where you can specify hyperparameters using the flags provided in options.py. The options just go in and manually change the dict in config.py afyer it is loaded so the two methods for setting the parameters are equivalent.
